### PR TITLE
Import packages from F-Droid

### PR DESCRIPTION
**Pull request by btry**
This PR connects the backend to F-Droid or a F-Droid like applications repository. 

The backend downloads once per day  the XML file describing all applications available and updates a local copy if this list in database. The admin can view the applications in the UI and import them to a new package. Once avaiable as a package, the app may be analyzed as usual, and deployed.

### CODE REVIEW

**Comment by ajsb85** [comment]
Google Play provides several link formats that let you bring users to your products in the way you want, from Android apps, web pages, ads, reviews, articles, social media posts, and more.

https://developer.android.com/distribute/marketing-tools/linking-to-google-play.html

**Comment by ajsb85** [comment]
This branch has conflicts that must be resolved

**Review by ajsb85 — DISMISSED** [review]
Reviewed

**Comment by DIOHz0r** [comment]
This issue should be related with #37, shouldn't it?

**Comment by ajsb85** [comment]
Hi, @btry 
317fb36 `fix rebase` is a commit message to fix.

**Comment by btry** [comment]
Hi @ajsb85 Just let the bot warn about that. I'll review the commit messages when the PR is ready to merge.

**Comment by DIOHz0r** [comment]
This PR needs a rebase to solve the conflicts and be up to date with the develop branch.

**Comment by btry** [comment]
Some screenshots of the progress

## features

* declare repos using the F-Droid app market
* download (with cron or by hands) the apps list of the market
* download an application and save it as a new package (cron task easily triggered with an import button)

image

image

image

**Comment by btry** [comment]
Needs some more work. A mini search in the fdroid apps tab is required.

File: inc/package.class.php [hunk 1]
```diff
@@ -190,7 +194,7 @@ public function prepareInputForAdd($input) {
       }
       try {
          $uploadedFile = $preparedFile['uploadedFile'];
-         $input['filename'] = 'flyvemdm/package/' . $this->fields['entities_id'] . '/' . uniqid() . '_' . basename($uploadedFile);
+         $input['filename'] = 'flyvemdm/package/' . $input['entities_id'] . '/' . uniqid() . '_' . basename($uploadedFile);
```
Comment (btry) [inline inc/package.class.php]: not related to fdroid: when creating a package, $this->fields should be empty. There is an error here.

File: inc/package.class.php [hunk 2]
```diff
@@ -349,11 +363,6 @@ protected function createEntityDirectory($dir) {
    public function getSearchOptionsNew() {
       $tab = parent::getSearchOptionsNew();
 
-      $tab[0] = [
-         'id'   => 'common',
-         'name' => __s('Package', 'flyvemdm'),
-      ];
-
```
Comment (btry) [inline inc/package.class.php]: this is useless, colling the parent implementation does the job

**Review by ajsb85 — DISMISSED** [review]

**Comment by ajsb85** [comment]
Hi, @btry 
There is an invalid commit message
`Merge branch 'release/2.0.0' into feature/fdroid`

**Comment by btry** [comment]
Hi 

This is the same message when you click on update branch on github :)

**Comment by btry** [comment]
I prepared a debase anyway, this will disappear soon

**Comment by btry** [comment]
closing, see #698

**Pull request closed without merging**